### ISSUE TITLE
Issue with Menu Items via the Admin Panel - Adding More Than One Child Item Not Working

### ISSUE THREAD

**Issue by smanny** [OrchardCMS/OrchardCore#15727]
When trying to add more than one child item under a parent menu item then saving it reverts back to its original position in Admin.

I of course need to be able to add more than one item in a dropdown menu.

Examining in the Inspection window, it seems that maybe the Javascript isn't assigning the correct attribute values. See attached screenshot. (It may appear from the screenshot that it worked, but as soon as I "Publish" it reverts the position of the second child item back to the top level)

<img width="1468" alt="Screenshot 2024-04-10 at 6 44 10 PM" src="https://github.com/OrchardCMS/OrchardCore/assets/7737516/c8dd67ad-036f-40e6-b247-28b01a8c7929">

**Comment by MikeAlhayek** [comment]
@smanny did you try this with the latest preview or main? I know this was an issue that I fixed couple weeks ago.

**Comment by smanny** [comment]
@MikeAlhayek I got this from main. Which files were updated?

**Comment by MikeAlhayek** [comment]
I just tested the admin menu from the main branch and it seems to be working. Not sure what am I missing. Can you check your log files for additional clues?

Did you follow the [1.9.0](https://docs.orchardcore.net/en/latest/docs/releases/1.9.0/) breaking changes log?

![image](https://github.com/OrchardCMS/OrchardCore/assets/24724371/b80ac551-5af6-4c99-a90a-1d96368f782c)

**Comment by smanny** [comment]
@MikeAlhayek Can the s, b, & c items be placed on the same level in your environment? I believe what you show there will output each child item under a new nested `<ul>`.

**Comment by smanny** [comment]
@MikeAlhayek The issue I am referring to is if you were to try and put (from your example) s, b, c, d on the same level. Right now, b is a child of s. C is a child of b etc. Are you able on your end to put those s, b, c, and d links as siblings? The outcome would create a dropdown under About with four child options (no flyouts). See my original attached image. When I try to put test 1 and test 2 together as siblings under Contact and publish, the second (or more) siblings will revert back to the top level. The system cannot save more than one sibling item together which would create an output like:
```
<ul>
    <li>About
        <ul>
            <li>s</li>
            <li>b</li>
            <li>c</li>
        </ul>
    </li>
</ul>
```
Rather than like:
```
<ul>
    <li>About
        <ul>
            <li>s
                <ul>
                     <li>b
                         <ul>
                             <li>c</li>
                         </ul>
                     </li>
               </ul>
            </li>
        </ul>
    </li>
</ul>
```

**Comment by MikeAlhayek** [comment]
@smanny The issue only happens with you have more than one child and try to move them out of their current parent to a main node. The driver actually throws an exception.

I was able to re-produce it and fix it in PR #15766 

```
OrchardCore.Menu.Drivers.MenuPartDisplayDriver by InvalidOperationException System.InvalidOperationException: The node already has a parent.
   at System.Text.Json.ThrowHelper.ThrowInvalidOperationException_NodeAlreadyHasParent()
   at System.Text.Json.Nodes.JsonObject.SetItem(String propertyName, JsonNode value)
   at OrchardCore.Menu.Drivers.MenuPartDisplayDriver.ProcessItem(MenuItemsListPart originalItems, JsonObject item) in C:\Code\OrchardCMS\OrchardCore\src\OrchardCore.Modules\OrchardCore.Menu\Drivers\MenuPartDisplayDriver.cs:line 124
   at OrchardCore.Menu.Drivers.MenuPartDisplayDriver.UpdateAsync(MenuPart part, IUpdateModel updater) in C:\Code\OrchardCMS\OrchardCore\src\OrchardCore.Modules\OrchardCore.Menu\Drivers\MenuPartDisplayDriver.cs:line 81
   at OrchardCore.ContentManagement.Display.ContentDisplay.ContentPartDisplayDriver`1.OrchardCore.ContentManagement.Display.ContentDisplay.IContentPartDisplayDriver.UpdateEditorAsync(ContentPart contentPart, ContentTypePartDefinition typePartDefinition, UpdateEditorContext context) in C:\Code\OrchardCMS\OrchardCore\src\OrchardCore\OrchardCore.ContentManagement.Display\ContentDisplay\ContentPartDisplayDriverTPart.cs:line 240
   at OrchardCore.ContentManagement.Display.ContentItemDisplayCoordinator.<>c.<<UpdateEditorAsync>b__9_0>d.MoveNext() in C:\Code\OrchardCMS\OrchardCore\src\OrchardCore\OrchardCore.ContentManagement.Display\ContentDisplay\ContentItemDisplayCoordinator.cs:line 347
```

**Comment by smanny** [comment]
@MikeAlhayek Thank you much!
